src/app/services/api/index.tsx

```diff
@@ -0,0 +1,59 @@
+import React, {FC, PropsWithChildren, useContext, useMemo} from 'react';
+import axios, {AxiosInstance, AxiosRequestConfig} from 'axios';
+import Config from 'react-native-config';
+
+import AsyncStore from '../asyncStorage';
+import UserContext from '../../context/user.context';
+import {googleSignOut} from '../auth/google.auth';
+
+export const axiosInstance: AxiosInstance = axios.create({
+  baseURL: Config.API_BASE_URL,
+  headers: {
+    'Content-Type': 'application/json',
+  },
+});
+
+export const Interceptor: FC<PropsWithChildren> = ({children}) => {
+  const [, setUserContext] = useContext(UserContext);
+
+  useMemo(() => {
```

Comment:
Should we use useEffect instead ?

and `return => axiosInstance.interceptors.response.eject` in use effect to release interceptor.

Reply:
👍🏻